App/CourseApp/Views/UIKit/HomeViewController.swift

```diff
@@ -10,26 +10,89 @@ import os
 import SwiftUI
 import UIKit
 
+// MARK: - UIViewControllerRepresentable
+struct HomeView: UIViewControllerRepresentable {
+    func makeUIViewController(context: Context) -> HomeViewController {
+        HomeViewController()
+    }
+    
+    func updateUIViewController(_ uiViewController: HomeViewController, context: Context) {
+    }
+}
+
 final class HomeViewController: UIViewController {
-    // swiftlint:disable prohibited_interface_builder
+    // swiftlint:disable:next prohibited_interface_builder
     @IBOutlet private var categoriesCollectionView: UICollectionView!
-    // swiftlint:enable prohibited_interface_builder
-    let logger = Logger()
+    
     // MARK: - DataSource
-    private lazy var dataProvider = MockDataProvider()
+    private let dataProvider = MockDataProvider()
     typealias DataSource = UICollectionViewDiffableDataSource<SectionData, [Joke]>
     typealias Snapshot = NSDiffableDataSourceSnapshot<SectionData, [Joke]>
     private lazy var dataSource = makeDataSource()
     private lazy var cancellables = Set<AnyCancellable>()
+    private let logger = Logger()
     
     override func viewDidLoad() {
         super.viewDidLoad()
         setup()
+        title = "Categories"
         // Do any additional setup after loading the view.
     }
+}
+
+// MARK: - UICollectionViewDelegateFlowLayout
+extension HomeViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let width = collectionView.bounds.width - UIConstants.cellSpacing
+        let height: CGFloat = collectionView.bounds.height / UIConstants.sectionScale
+        return CGSize(width: width, height: height)
+    }
+}
+
+// MARK: - UICollectionViewDelegate
+extension HomeViewController: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        self.logger.info("Home collection view did select item at \(indexPath)")
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, willDisplay cell: UICollectionViewCell, forItemAt indexPath: IndexPath) {
+        self.logger.info("Home collection view cell will display \(indexPath)")
+    }
+}
+
+// MARK: - UISetup
+private extension HomeViewController {
+    func setup() {
+        setupCollectionView()
+        readData()
+    }
     
+    func setupCollectionView() {
+        categoriesCollectionView.register(HorizontalScrollingImageCell.self)
+        categoriesCollectionView.register(LabelCollectionViewCell.self, forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader)
```

Comment:
Let get rid of UIKit here the swiftUI label cell as hostingConfiguration can work here better